src/label.js

```diff
@@ -0,0 +1,68 @@
+var temp_div = null;
+
+function addLabels(text, target){
+  console.log('adding label');
+  var num_labels = d3.select(target.parentNode).selectAll(".label").size();
+  // If a label already exists
+  if (num_labels > 0){
+    // Update the text 
+    text = d3.select(target.parentNode).select(".label").text();
+    // Delete the inside node
+    d3.select(target.parentNode).select(".label").remove();
+  }
+
+  // Adding an editable div outside
+  var container = document.getElementById("d3_container");
+  var label = document.createElement("div");
+  label.appendChild(document.createTextNode(text));
+  label.style.position = "absolute";
+  label.setAttribute("z-index", "1");
+  var name = d3.select(target.parentNode).attr("class").split(" ")[0];
+  if (name == "node"){
+    var translation = target.parentNode.getAttribute("transform");
+    translation = translation.substring(9).replace("(", "").replace(")","").split(",");
+    var cx = translation[0]-10;
+    var cy = translation[1]-10;
+  }else{
```

Comment:
Else should only be used for a default case, but it looks like its being used for a specific one